### PR TITLE
ACSS-41058: XDM Registry: add new attributes for inline messages

### PR DESCRIPTION
**Pull request by mukullamba04**
Wiki: https://wiki.corp.adobe.com/display/voyager/Changes+to+come+Jan+2022+and+later

### CODE REVIEW

File: extensions/adobe/experience/journeyOrchestration/stepEvents/journeyStepEventActionExecutionFieldsMixin.schema.json [hunk 1]
```diff
@@ -123,6 +123,73 @@
...
+                "type": "string"
+            },
+            "https://ns.adobe.com/experience/journeyOrchestration/surfaceID": {
+                "title": "SurfaceID",
+                "description": "Surface ID",
+                "type": "string"
+            },
+            "https://ns.adobe.com/experience/journeyOrchestration/channel": {
+                "title": "Channel",
+                "description": "Channel Name",
```
Comment (saurabh111091) [inline extensions/adobe/experience/journeyOrchestration/stepEvents/journeyStepEventActionExecutionFieldsMixin.schema.json:150]: Can we add some examples in the description ?

File: extensions/adobe/experience/journeyOrchestration/stepEvents/journeyStepEventActionExecutionFieldsMixin.schema.json [hunk 2]
```diff
@@ -123,6 +123,73 @@
...
+            },
+
+            "https://ns.adobe.com/experience/journeyOrchestration/actionGroupID": {
+                "title": "ActionGroupID",
+                "description": "Action Group ID",
+                "type": "string"
+            },
+            "https://ns.adobe.com/experience/journeyOrchestration/actionGroupName": {
+                "title": "ActionGroupName",
+                "description": "Action Group Name",
```
Comment (saurabh111091) [inline extensions/adobe/experience/journeyOrchestration/stepEvents/journeyStepEventActionExecutionFieldsMixin.schema.json:166]: Can we add some examples in the description ?

File: extensions/adobe/experience/journeyOrchestration/stepEvents/journeyStepEventActionExecutionFieldsMixin.schema.json [hunk 3]
```diff
@@ -123,6 +123,73 @@
...
+                "type": "string"
+            },
+            "https://ns.adobe.com/experience/journeyOrchestration/experimentationID": {
+                "title": "ExperimentationID",
+                "description": "Experimentation ID",
+                "type": "string"
+            },
+            "https://ns.adobe.com/experience/journeyOrchestration/experimentationName": {
+                "title": "ExperimentationName",
+                "description": "Experimentation Name",
```
Comment (saurabh111091) [inline extensions/adobe/experience/journeyOrchestration/stepEvents/journeyStepEventActionExecutionFieldsMixin.schema.json:176]: Can we add some examples in the description ?

File: extensions/adobe/experience/journeyOrchestration/stepEvents/journeyStepEventActionExecutionFieldsMixin.schema.json [hunk 4]
```diff
@@ -123,6 +123,73 @@
...
+                "type": "string"
+            },
+            "https://ns.adobe.com/experience/journeyOrchestration/treatmentID": {
+                "title": "TreatmentID",
+                "description": "Treatment ID applied to execute the action",
+                "type": "string"
+            },
+            "https://ns.adobe.com/experience/journeyOrchestration/treatmentName": {
+                "title": "TreatmentName",
+                "description": "Treatment name applied to execute the action",
```
Comment (saurabh111091) [inline extensions/adobe/experience/journeyOrchestration/stepEvents/journeyStepEventActionExecutionFieldsMixin.schema.json:186]: Can we add some examples in the description ?

**Review by saurabh111091 — APPROVED** [review]
+1 
But will be great if we can add some examples in description of some of the fields.

**Review by lecercle95 — APPROVED** [review]

File: extensions/adobe/experience/journeyOrchestration/stepEvents/journeyStepEventActionExecutionFieldsMixin.schema.json (hunk 4 above)
Comment (mukullamba04) [reply extensions/adobe/experience/journeyOrchestration/stepEvents/journeyStepEventActionExecutionFieldsMixin.schema.json:186]: As discussed with Stephane, TreatmentName will be a user-created value on UI and hence I think we should be good without example here. User will know the treatment name created by them

File: extensions/adobe/experience/journeyOrchestration/stepEvents/journeyStepEventActionExecutionFieldsMixin.schema.json (hunk 3 above)
Comment (mukullamba04) [reply extensions/adobe/experience/journeyOrchestration/stepEvents/journeyStepEventActionExecutionFieldsMixin.schema.json:176]: As discussed with Stephane, ExperimentationName will be a user-created value on UI and hence I think we should be good without example here. User will know the experimentation name created by them

File: extensions/adobe/experience/journeyOrchestration/stepEvents/journeyStepEventActionExecutionFieldsMixin.schema.json (hunk 2 above)
Comment (mukullamba04) [reply extensions/adobe/experience/journeyOrchestration/stepEvents/journeyStepEventActionExecutionFieldsMixin.schema.json:166]: As discussed with Stephane, ActionGroupName will be a user-created value on UI and hence I think we should be good without example here. User will know the action group name created by them

**Review by lecercle95 — APPROVED** [review]

File: extensions/adobe/experience/journeyOrchestration/stepEvents/journeyStepEventActionExecutionFieldsMixin.schema.json [hunk 5]
```diff
@@ -123,6 +123,105 @@
...
+                "title": "MessageID",
+                "description": "Message ID referenced in inline message action of a journey",
+                "type": "string"
+            },
+            "https://ns.adobe.com/experience/journeyOrchestration/publishedMessageID": {
+                "title": "PublishedMessageID",
+                "description": "Published message ID in inline message action of a journey",
+                "type": "string"
+            },
+            "https://ns.adobe.com/experience/journeyOrchestration/interactionType": {
```
Comment (mjasrotia) [inline extensions/adobe/experience/journeyOrchestration/stepEvents/journeyStepEventActionExecutionFieldsMixin.schema.json:138]: If we know they can be either of "transactional" or "marketing", can we add these as enums ? That way you can ensure the right values flowing into this property and design your downstream flows accordingly

File: extensions/adobe/experience/journeyOrchestration/stepEvents/journeyStepEventActionExecutionFieldsMixin.schema.json [hunk 6]
```diff
@@ -123,6 +123,105 @@
...
+                "title": "InteractionType",
+                "description": "Indicate the interaction type, either 'marketing' or 'transactional'. Remark: this will be relevant for message and custom actions",
+                "type": "string"
+            },
+            "https://ns.adobe.com/experience/journeyOrchestration/surfaceID": {
+                "title": "SurfaceID",
+                "description": "Surface ID",
+                "type": "string"
+            },
+            "https://ns.adobe.com/experience/journeyOrchestration/channel": {
```
Comment (mjasrotia) [inline extensions/adobe/experience/journeyOrchestration/stepEvents/journeyStepEventActionExecutionFieldsMixin.schema.json:148]: Same as above...enum ?

File: extensions/adobe/experience/journeyOrchestration/stepEvents/journeyStepEventActionExecutionFieldsMixin.schema.json (hunk 5 above)
Comment (mukullamba04) [reply extensions/adobe/experience/journeyOrchestration/stepEvents/journeyStepEventActionExecutionFieldsMixin.schema.json:138]: Hey @mjasrotia , thanks for reviewing. We discussed it in our team and it does make sense to eventually make it enum, but for now it is too early and we would want to keep it like this for sometime as eventually the naming might be a bit different when hardened and more values could be added.

File: extensions/adobe/experience/journeyOrchestration/stepEvents/journeyStepEventActionExecutionFieldsMixin.schema.json (hunk 6 above)
Comment (mukullamba04) [reply extensions/adobe/experience/journeyOrchestration/stepEvents/journeyStepEventActionExecutionFieldsMixin.schema.json:148]: Hey @mjasrotia , thanks for reviewing. We discussed it in our team and it does make sense to eventually make it enum, but for now it is too early and we would want to keep it like this for sometime as eventually the naming might be a bit different when hardened and more values could be added.

File: extensions/adobe/experience/journeyOrchestration/stepEvents/journeyStepEventActionExecutionFieldsMixin.schema.json (hunk 5 above)
Comment (mjasrotia) [reply extensions/adobe/experience/journeyOrchestration/stepEvents/journeyStepEventActionExecutionFieldsMixin.schema.json:138]: @mukullamba04 , I understand that but please be aware of the fact that any enums added after the schemas are released and data is ingested could be a breaking change and that is not allowed

Comment (mukullamba04) [reply extensions/adobe/experience/journeyOrchestration/stepEvents/journeyStepEventActionExecutionFieldsMixin.schema.json:138]: > @mukullamba04 , I understand that but please be aware of the fact that any enums added after the schemas are released and data is ingested could be a breaking change and that is not allowed

As discussed with @mjasrotia , we will be sending test data for now and will change data type, if we do, before ingesting live customer data.

**Review by mjasrotia — APPROVED** [review]

**Review by prabhum2 — APPROVED** [review]

**Pull request merged**